go-client/pkg/bucketclient/bucketclienterror.go

```diff
@@ -0,0 +1,32 @@
+package bucketclient
+
+import (
+	"fmt"
+)
+
+type BucketClientError struct {
+	ApiMethod string
+	HttpMethod string
+	Endpoint string
+	Resource string
+	StatusCode int
+	ErrorType string
+	Err error
```

Comment:
since it hold another error, it should probably also implement the `Unwrap` method to get access to that internal type.